messages2/src/decorators/thread.rs

```diff
@@ -0,0 +1,96 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::maybe_known::MaybeKnown;
+
+/// Struct representing the `~thread` decorator from its [RFC](https://github.com/hyperledger/aries-rfcs/blob/main/concepts/0008-message-id-and-threading/README.md).
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct Thread {
+    pub thid: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub pthid: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub sender_order: Option<u32>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub received_orders: Option<HashMap<String, u32>>, // should get replaced with DID.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub goal_code: Option<MaybeKnown<ThreadGoalCode>>,
+}
+
+impl Thread {
+    pub fn new(thid: String) -> Self {
+        Self {
+            thid,
+            pthid: None,
+            sender_order: None,
+            received_orders: None,
+            goal_code: None,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
```

Comment:
Maybe i'm missing something, but should these well-known goal codes be serde renamed so that they serialize/deserialize correctly for the RFC format? e.g. `aries.vc` etc; https://github.com/hyperledger/aries-rfcs/blob/main/concepts/0519-goal-codes/README.md#well-known-goal-codes

Reply:
Yeah, so... This used to have a variant `Other(String)` which would accept other variants, but removed it in favor of the `MaybeKnown` wrapper because there were a lot of constants involved and boilerplate code. And, of course, I forgot to add the renames to keep everything in check. I'll have another look at types like this to make sure they're implemented properly. 

Great catch, George!

Reply:
Fixed.